BackEndApp/deployments/models.py

```diff
@@ -86,14 +87,22 @@ def download_app(self):
 
                 return response
             else:
-                status, success, message = (
-                    200,
-                    True,
-                    "Executable Downloaded Successfully",
-                )
-                return JsonResponse(
-                    {"success": success, "message": message}, status=status
+                app = os.path.join(self.deployment_dir, "app.py")
+                dist_path = os.path.abspath('dist')
+                PyInstaller.__main__.run([
+                    app,
+                    "--onefile",
+                    "--clean",
+                ])
+                response = HttpResponse(
+                    open(os.path.join(dist_path, 'app.exe'), "rb"),
+                    headers={
+                        "Content-Type": "application/vnd.microsoft.portable-executable",
+                        "Content-Disposition": "attachment; filename=app.exe",
+                    },
```

Comment:
@Aditya-ds-1806 could you post the file size of the `app.exe`?

Reply:
I was concerned about the size, comes close to 80mb.

Reply:
Ouch okay. Let's see if we can bring that down to a couple of MBs that would be great.

Reply:
We can take a look at UPX maybe? But then we will have to install the utility on the server as well.

Reply:
Oh that might be a good option. Just try that see what the final file size is

Reply:
Also add the support for a linux binary as well along with .exe

Reply:
If we build Linux binaries, how do we know which file the user is interested in? We just know that it is an executable

Reply:
We could have an option in the UI for the user to choose whatever he wants

Reply:
Having it in the UI is an option, but to improve the UX we could essentially check the Client info by using the **navigator** API as below:

```
let OSName="Unknown OS";

if (navigator.appVersion.indexOf("Win")!=-1) OSName="Windows";

if (navigator.appVersion.indexOf("Mac")!=-1) OSName="MacOS";

if (navigator.appVersion.indexOf("Linux")!=-1) OSName="Linux";
```

You could later send the `OSName` to the `BackEndApp` and accordingly handle creation of a Linux or Windows executable as required (without having to pester the user at all 🎉).

Reply:
I was thinking along similar lines, but then got worried about browser support.

Reply:
https://developer.mozilla.org/en-US/docs/Web/API/Navigator/appVersion
It seems to be deprecated

Reply:
Autodetecting the OS is good but the reason I didn't mention it so that the user has a choice (it's better to give them a choice). We could select the detected OS  as default.

Reply:
Agree with @RusherRG, seems to be more flexible according to me.

Reply:
I see... but apparently many still use it 🤔.
Let's see if we can find a better alternative, or we can always fallback to including an option in the Frontend.

Reply:
@RusherRG Tried UPX, brought the size down to 60MB.

Reply:
Ignore all of that, I was doing something stupid, it is 5.5 MB without TF and UPX applied.

Reply:
:joy: Wow that's good to hear. What was the mistake though?

Reply:
Well I don't really have a background in python. I thought I had setup a virtual environment but didn't realise that I need to activate it. So basically every single python package that was available globally got bundled and created this massive 90MB file 😂